FirebaseAppDistribution/Sources/FIRFADApiService.m

```diff
@@ -72,9 +75,13 @@ + (NSMutableURLRequest *)createHTTPRequest:(NSString *)method
   FIRFADInfoLog(@"Requesting releases for app id - %@", [[FIRApp defaultApp] options].googleAppID);
   [request setURL:[NSURL URLWithString:urlString]];
   [request setHTTPMethod:method];
-  [request setValue:authTokenResult.authToken forHTTPHeaderField:kInstallationAuthHeader];
-  [request setValue:[[FIRApp defaultApp] options].APIKey forHTTPHeaderField:kApiHeaderKey];
-  [request setValue:[NSBundle mainBundle].bundleIdentifier forHTTPHeaderField:kApiBundleKey];
+  [request setValue:authTokenResult.authToken forHTTPHeaderField:kFIRFADInstallationAuthHeader];
+  [request setValue:[[FIRApp defaultApp] options].APIKey forHTTPHeaderField:kFIRFADApiHeaderKey];
+  [request setValue:[NSBundle mainBundle].bundleIdentifier forHTTPHeaderField:kFIRFADApiBundleKey];
+  [request setValue:[FIRApp firebaseUserAgent] forHTTPHeaderField:kFIRFADFirebaseUserAgentKey];
+  [request setValue:@([FIRHeartbeatInfo heartbeatCodeForTag:kFIRFADHeartbeatTag]).stringValue
```

Comment:
`[FIRHeartbeatInfo heartbeatCodeForTag:]` makes disk operations, so it should not be called on main thread. It's hard to understand from the scope of the PR on what thread this method is called. Could you please check it.

Reply:
At the moment we leave whether this is on the main thread or a different thread to the caller (as I understand it) - if we need to make sure this is on a background thread, perhaps we should do that directly from this class.

Reply:
Then yes, I would recommend doing it directly.